### DISCUSSION TITLE
Postman requests: Machine is not connected yet.

### DISCUSSION THREAD

**Discussion by Eddie-Hartman** [Snapmaker/Luban#1168] [Q&A]
I'm trying to develop an app to check the status of my Snapmaker prints so that I do not have to go to the machine to check them.

After sending a connect post request, I am taking the token from the post request and sending a status get request with the token as a parameter, but I get a "Machine is not connected yet." response with a 401.

Onto the questions:
1. What repo is this response coming from? I grepped the Snapmaker Luban and the Controller repos, but do not see it.
2. I've allowed the connection on the machine by confirming, but still get this response. If I open Luban and then grab the token Luban uses, I can make a Postman status api request and get a real response, but if I connect in Postman it will always say the machine is not connected. What am I doing wrong? UPDATE: It seems like the expiration for the token is incredibly short, so the requests have to be sent in very rapid succession.

**Comment by parachvte** [comment]
Try `/api/v1/connect` API without parameter `token`, it will request for a new session (with new token).

**Reply by Eddie-Hartman** [reply]
Right. That is the first request I'm sending. I'm not entirely sure of the purpose of sending a connect request with a token as a parameter (I understand that this is how some apis may handle refreshing the token).

I was able to get it working reliably in Postman by making a collection runner that would send the status request immediately after the connect request, I grab the token from the response and send it in as a parameter to the status request. I was just surprised that the token expires in less than a second.

Is there any public documentation for this API I'm working with?
